bin/sdk/gutenberg.js

```diff
@@ -1,116 +1,30 @@
 /** @format */
-
 /**
  * External dependencies
  */
-const chalk = require( 'chalk' );
 const path = require( 'path' );
-const webpack = require( 'webpack' );
-const { isEmpty, omitBy } = require( 'lodash' );
 
-const __rootDir = path.resolve( __dirname, '../../' );
-const CopyWebpackPlugin = require( path.resolve(
+exports.config = ( {
+	argv: { editorScript, viewScript, outputDir, outputEditorFile, outputViewFile },
+	getBaseConfig,
 	__rootDir,
-	'server/bundler/copy-webpack-plugin'
-) );
-const getBaseConfig = require( path.join( __rootDir, 'webpack.config.js' ) );
-
-const omitPlugins = [
-	CopyWebpackPlugin,
-	webpack.HotModuleReplacementPlugin,
-];
-
-const wordpressPackages = [
-	'api-fetch',
-	'block-serialization-spec-parser',
-	'blocks',
-	'components',
-	'data',
-	'editor',
-	'element',
-	'i18n',
-];
-
-const getWordPressExternals = () =>
-	wordpressPackages.reduce( ( externals, package ) => {
-		externals[ `@wordpress/${ package }` ] = {
-			window: [
-				'wp',
-				// this is not as aggressive as `_.camelCase` in converting to
-				// uppercase, where Lodash will convert letters following numbers
-				package.replace(
-					/-([a-z])/g,
-					( match, letter ) => letter.toUpperCase()
-				)
-			]
-		};
-		return externals;
-	}, {} );
-
-const outputHandler = ( error, stats ) => {
-	if ( error ) {
-		console.error( error );
-		console.log( chalk.red( 'Failed to build Gutenberg extension' ) );
-		process.exit( 1 );
-	}
-
-	console.log( stats.toString() );
-
-	if ( stats.hasErrors() ) {
-		console.log( chalk.red( 'Finished building Gutenberg extension but with errors.' ) );
-	} else if ( stats.hasWarnings() ) {
-		console.log( chalk.yellow( 'Successfully built Gutenberg extension but with warnings.' ) );
-	} else {
-		console.log( chalk.green( 'Successfully built Gutenberg extension' ) );
-	}
-};
-
-exports.compile = args => {
-	const options = {
-		outputDir: path.join( path.dirname( args.editorScript ), 'build' ),
-		...args,
-	};
-
-	const name = path.basename( path.dirname( options.editorScript ).replace( /\/$/, '' ) );
-	const baseConfig = getBaseConfig( { extensionName: name } );
+} ) => {
+	const baseConfig = getBaseConfig( { externalizeWordPressPackages: true } );
+	const name = path.basename( path.dirname( editorScript ).replace( /\/$/, '' ) );
 
-	const config = {
+	return {
 		...baseConfig,
 		...{
 			context: __rootDir,
```

Comment:
Moving `context: __rootDir,` to `sdk-cli.js` would probably make sense since it needs to be there each time `sdk-cli.js` is run via bin scripts.

Thus we also don't need to pass `__rootDir` down to `gutenberg.js`

Reply:
thanks. I wondered about that but didn't feel confident enough on my own to move it

Reply:
We could even consider moving it to `webpack.config.js` if it's not too much noise there, although it's relevant only for the SDK CLI.

Reply:
moved!